R/ergm_tidiers.R

```diff
@@ -0,0 +1,198 @@
+                      exponentiate = FALSE, quick = FALSE, ...) {
+    if (quick) {
+        co <- x$coef
+        ret <- data.frame(term = names(co), estimate = unname(co))
+        return(ret)
+    }
+    co <- ergm::summary.ergm(x, ...)$coefs
+    
+    nn <- c("estimate", "std.error", "mcmc.error", "p.value")
+    if (inherits(co, "listof")) {
+        # multiple response variables
+        ret <- plyr::ldply(co, fix_data_frame, nn[1:ncol(co[[1]])],
+                           .id = "response")
+        ret$response <- stringr::str_replace(ret$response, "Response ", "")
+    } else {
+        ret <- fix_data_frame(co, nn[1:ncol(co)])
+    }
+    
+    process_ergm(ret, x, conf.int = conf.int, conf.level = conf.level,
+                 exponentiate = exponentiate)
+}
+
+#' @rdname ergm_tidiers
+#'
+#' @param deviance whether to report null and residual deviance for the model,
+#' along with degrees of freedom; defaults to \code{FALSE}
+#' @param mcmc whether to report MCMC interval, burn-in and sample size used to
+#' estimate the model; defaults to \code{FALSE}
+#' @param ... extra arguments passed to \code{\link[ergm]{summary.ergm}}
+#'
+#' @return \code{glance.ergm} returns a one-row data.frame with the columns
+#'   \item{independence}{Whether the model assumed dyadic independence}
+#'   \item{iterations}{The number of iterations performed before convergence}
+#'   \item{logLik}{If applicable, the log-likelihood associated with the model}
+#'   \item{AIC}{The Akaike Information Criterion}
+#'   \item{BIC}{The Bayesian Information Criterion}
+#'
+#' If \code{deviance=TRUE}, and if the model supports it, the
+#' data frame will also contain the columns
+#'   \item{null.deviance}{The null deviance of the model}
+#'   \item{df.null}{The degrees of freedom of the null deviance}
+#'   \item{residual.deviance}{The residual deviance of the model}
+#'   \item{df.residual}{The degrees of freedom of the residual deviance}
+#'
+#' Last, if \code{mcmc=TRUE}, the data frame will also contain
+#' the columns
+#'   \item{MCMC.interval}{The interval used during MCMC estimation}
+#'   \item{MCMC.burnin}{The burn-in period of the MCMC estimation}
+#'   \item{MCMC.samplesize}{The sample size used during MCMC estimation}
+#'
+#' @export
+glance.ergm <- function(x, deviance = FALSE, mcmc = FALSE, ...) {
+    # will show appropriate warnings about standard errors, pseudolikelihood etc.
+    s <- ergm::summary.ergm(x, ...)
+    # dyadic (in)dependence and number of MCMLE iterations
+    ret <- data.frame(independence = s$independence, iterations = x$iterations)
+    # log-likelihood
+    ret$logLik <- tryCatch(as.numeric(ergm::logLik.ergm(x)), error = function(e) NULL)
+    # null and residual deviance
+    s <- s$devtable
+    if (deviance & length(s) > 0) {
+        # dirty, but recomputing the logLik table would require much more effort
+        # that would add a dependency to the network package (for the dyad count)
```

Comment:
The 'real' way to compute the following statistics requires [counting the number of dyads](https://github.com/cran/ergm/blob/master/R/summary.ergm.R#L122) in the network used by the model. This would introduce a dependency to the `network` package, so I went for the ugly hack of the deviance table instead.